### PR TITLE
[TIMOB-19939] Deprecating KEYBOARD_ for KEYBOARD_TYPE

### PR DESCRIPTION
**Pull request by AngelkPetkov**
Also deprecated appearance in favour for keyboardAppearance in TextField and TextArea.

:JIRA: https://jira.appcelerator.org/browse/TIMOB-19939?filter=-1

:JIRA: https://jira.appcelerator.org/browse/TIMOB-20000?filter=-1
Demo Code : 

```
Ti.UI.setBackgroundColor('white');
var win = Ti.UI.createWindow({
  title: 'Click window to test'
});

var field = Ti.UI.createTextArea({
  width           : 300,
  height          : 50,
  top             : 200,
  right           : 0 ,
  backgroundColor : "#f0f0f0",
  value           : "Hello",
  keyboardType : Titanium.UI.KEYBOARD_ASCII,
  appearance: Titanium.UI.KEYBOARD_APPEARANCE_LIGHT,
});


 var textField = Ti.UI.createTextField({
    borderStyle   : Ti.UI.INPUT_BORDERSTYLE_ROUNDED,
  color           : '#336699',
  top             : 100, 
  right           : 0,
  width           : 250, 
  height          : 60,
  backgroundColor :'Green',
  keyboardType    : Titanium.UI.KEYBOARD_TWITTER,
  appearance      : Titanium.UI.KEYBOARD_APPEARANCE_DARK
});

var search = Titanium.UI.createSearchBar({
    barColor            :'#000', 
    showCancel          :true,
    height              :43,
    top                 :20,
    left                :0,
    keyboardType        : Titanium.UI.KEYBOARD_TYPE_URL,
    keyboardAppearance  : Titanium.UI.KEYBOARD_APPEARANCE_DARK
});
field.autocorrect = false;

win.add(textField);
win.add(field);
win.add(search);
win.open();
```

### CODE REVIEW

File: apidoc/Titanium/UI/TextArea.yml [hunk 1]
```diff
@@ -72,9 +72,19 @@ properties:
     constants: Titanium.UI.KEYBOARD_APPEARANCE_*
     default: <Titanium.UI.KEYBOARD_APPEARANCE_DEFAULT>
     platforms: [iphone, ipad]
-    
+    deprecated:
+        since: "5.2.0"
+        notes: Use keyboardAppearance
```
Comment (hansemannn) [inline apidoc/Titanium/UI/TextArea.yml]: Replace with: "Use `keyboardAppearance` instead."

File: apidoc/Titanium/UI/TextField.yml [hunk 2]
```diff
@@ -75,7 +75,17 @@ properties:
     constants: Titanium.UI.KEYBOARD_APPEARANCE_*
     default: <Titanium.UI.KEYBOARD_APPEARANCE_DEFAULT>
     platforms: [iphone, ipad]
-    
+    deprecated:
+        since: "5.2.0"
+        notes: Use keyboardAppearance.
```
Comment (hansemannn) [inline apidoc/Titanium/UI/TextField.yml]: Replace with: "Use `keyboardAppearance` instead."

File: apidoc/Titanium/UI/UI.yml [hunk 3]
```diff
@@ -1447,6 +1447,9 @@ properties:
     type: Number
     permission: read-only
     platforms: [android, iphone, ipad]
+    deprecated:
+        since: "5.2.0"
+        notes: Use KEYBOARD_TYPE_ASCII.
```
Comment (hansemannn) [inline apidoc/Titanium/UI/UI.yml]: Replace with: "Use `KEYBOARD_TYPE_ASCII` instead."

Other constants, too.

File: apidoc/Titanium/UI/UI.yml [hunk 4]
```diff
@@ -1521,8 +1542,108 @@ properties:
         `KEYBOARD_DEFAULT`.
     type: Number
     permission: read-only
+    deprecated:
+        since: "5.2.0"
+        notes: Use KEYBOARD_TYPE_PHONE_PAD
+
+  - name: KEYBOARD_URL
```
Comment (hansemannn) [inline apidoc/Titanium/UI/UI.yml]: Not `KEYBOARD_TYPE_URL` ?

File: apidoc/Titanium/UI/UI.yml [hunk 5]
```diff
@@ -1521,8 +1542,108 @@ properties:
...
+        since: "5.2.0"
+        notes: Use KEYBOARD_TYPE_URL
+
+  - name: KEYBOARD_TYPE_DECIMAL_PAD
+    summary: Use a keyboard with decimal numbers only, with the pad keyboard layout.
+    description: |
+        Use with the <Titanium.UI.TextField.keyboardType> and <Titanium.UI.TextArea.keyboardType>
+        properties.
+
+        On iPad, this behaves the same as <Titanium.UI.KEYBOARD_DEFAULT>.
```
Comment (hansemannn) [inline apidoc/Titanium/UI/UI.yml]: Not `Titanium.UI.KEYBOARD_TYPE_DEFAULT`?

File: apidoc/Titanium/UI/UI.yml [hunk 6]
```diff
@@ -1521,8 +1542,108 @@ properties:
...
+    permission: read-only
+
+  - name: KEYBOARD_TYPE_EMAIL
+    summary: Use a keyboard suitable for composing email, with the standard keyboard layout.
+    description: |
+        Use with the <Titanium.UI.TextField.keyboardType> and <Titanium.UI.TextArea.keyboardType>
+        properties.
+
+        Note: on Mobile Web, if you enable `passwordType`, the `keyboardType` value will be
+        overridden with `KEYBOARD_DEFAULT`.
```
Comment (hansemannn) [inline apidoc/Titanium/UI/UI.yml]: Not `KEYBOARD_TYPE_DEFAULT`?

File: apidoc/Titanium/UI/UI.yml [hunk 7]
```diff
@@ -1521,8 +1542,108 @@ properties:
...
+    permission: read-only
+    platforms: [android, iphone, ipad]
+
+  - name: KEYBOARD_TYPE_NUMBER_PAD
+    summary: Use a keyboard with a number pad only, with the pad keyboard layout.
+    description: |
+        Use with the <Titanium.UI.TextField.keyboardType> and <Titanium.UI.TextArea.keyboardType>
+        properties.
+
+        On iPad, this behaves the same as <Titanium.UI.KEYBOARD_NUMBERS_PUNCTUATION>.
```
Comment (hansemannn) [inline apidoc/Titanium/UI/UI.yml]: Not `Titanium.UI.KEYBOARD_TYPE_NUMBERS_PUNCTUATION `?

File: apidoc/Titanium/UI/UI.yml [hunk 8]
```diff
@@ -1521,8 +1542,108 @@ properties:
...
+
+        On Mobile Web, if you enable `passwordType`, the `keyboardType` value will be overridden with
+        `KEYBOARD_DEFAULT`.
```
Comment (hansemannn) [inline apidoc/Titanium/UI/UI.yml]: Not `KEYBOARD_TYPE_DEFAULT`?

File: apidoc/Titanium/UI/UI.yml [hunk 9]
```diff
@@ -1521,8 +1542,108 @@ properties:
...
+    permission: read-only
 
-  - name: KEYBOARD_WEBSEARCH
+  - name: KEYBOARD_TYPE_PHONE_PAD
+    summary: Use a keyboard with a phone-style number pad, with the pad keyboard layout.
+    description: |
+        Use with the <Titanium.UI.TextField.keyboardType> and <Titanium.UI.TextArea.keyboardType>
+        properties.
+
+        On iPad, this behaves the same as <Titanium.UI.KEYBOARD_NUMBERS_PUNCTUATION>.
```
Comment (hansemannn) [inline apidoc/Titanium/UI/UI.yml]: Not `Titanium.UI.KEYBOARD_TYPE_NUMBERS_PUNCTUATION`?

File: apidoc/Titanium/UI/UI.yml [hunk 10]
```diff
@@ -1521,8 +1542,108 @@ properties:
...
+
+        On Mobile Web, if you enable `passwordType`, the `keyboardType` value will be overridden with
+        `KEYBOARD_DEFAULT`.
```
Comment (hansemannn) [inline apidoc/Titanium/UI/UI.yml]: Not `KEYBOARD_TYPE_DEFAULT`?

File: apidoc/Titanium/UI/UI.yml [hunk 11]
```diff
@@ -1544,27 +1665,27 @@ properties:
...
+  - name: KEYBOARD_TYPE_URL
     summary: Use a keyboard optimized for entering URLs, with the standard keyboard layout.
     description: |
-        Use with the <Titanium.UI.TextField.keyboardType> and <Titanium.UI.TextArea.keyboardType> 
+        Use with the <Titanium.UI.TextField.keyboardType> and <Titanium.UI.TextArea.keyboardType>
         properties.
-        
+
         On Mobile Web, if you enable `passwordType`, the `keyboardType` value will be overridden with
         `KEYBOARD_DEFAULT`.
```
Comment (hansemannn) [inline apidoc/Titanium/UI/UI.yml]: Not `KEYBOARD_TYPE_DEFAULT`?

File: apidoc/Titanium/UI/UI.yml [hunk 12]
```diff
@@ -1544,27 +1665,27 @@ properties:
...
-        <Titanium.UI.LANDSCAPE_RIGHT>, 
-        <Titanium.UI.FACE_UP>, 
-        <Titanium.UI.FACE_DOWN>, 
+        <Titanium.UI.PORTRAIT>,
+        <Titanium.UI.UPSIDE_PORTRAIT>,
+        <Titanium.UI.LANDSCAPE_LEFT>,
+        <Titanium.UI.LANDSCAPE_RIGHT>,
+        <Titanium.UI.FACE_UP>,
+        <Titanium.UI.FACE_DOWN>,
         and <Titanium.UI.UNKNOWN>.
```
Comment (hansemannn) [inline apidoc/Titanium/UI/UI.yml]: Restore thus part is it is just whitespacing.

File: iphone/Classes/UIModule.m [hunk 13]
```diff
@@ -128,17 +128,29 @@ -(NSNumber*)RETURNKEY_CONTINUE
...
+
+MAKE_SYSTEM_PROP_DEPRECATED_REPLACED(KEYBOARD_DEFAULT,UIKeyboardTypeDefault, @"Titanium.UI.KEYBOARD_DEFAULT", @"5.2.0", @"Titanium.UI.KEYBOARD_TYPE_DEFAULT");
+MAKE_SYSTEM_PROP_DEPRECATED_REPLACED(KEYBOARD_URL,UIKeyboardTypeURL, @"Titanium.UI.KEYBOARD_URL", @"5.2.0", @"Titanium.UI.KEYBOARD_TYPE_URL");
+MAKE_SYSTEM_PROP_DEPRECATED_REPLACED(KEYBOARD_ASCII,UIKeyboardTypeASCIICapable, @"Titanium.UI.KEYBOARD_ASCII", @"5.2.0", @"Titanium.UI.KEYBOARD_TYPE_ASCII");
+MAKE_SYSTEM_PROP_DEPRECATED_REPLACED(KEYBOARD_NUMBERS_PUNCTUATION,UIKeyboardTypeNumbersAndPunctuation, @"Titanium.UI.KEYBOARD_NUMBERS_PUNCTUATION", @"5.2.0", @"Titanium.UI.KEYBOARD_TYPE_NUMBERS_PUNCTUATION");
+MAKE_SYSTEM_PROP_DEPRECATED_REPLACED(KEYBOARD_NUMBER_PAD,UIKeyboardTypeNumberPad, @"Titanium.UI.KEYBOARD_NUMBER_PAD", @"5.2.0", @"Titanium.UI.KEYBOARD_TYPE_NUMBER_PAD");
+MAKE_SYSTEM_PROP_DEPRECATED_REPLACED(KEYBOARD_DECIMAL_PAD,UIKeyboardTypeDecimalPad, @"Titanium.UI.KEYBOARD_DECIMAL_PAD", @"5.2.0", @"Titanium.UI.KEYBOARD_TYPE_DECIMAL_PAD");
+MAKE_SYSTEM_PROP_DEPRECATED_REPLACED(KEYBOARD_PHONE_PAD,UIKeyboardTypeNamePhonePad, @"Titanium.UI.KEYBOARD_PHONE_PAD", @"5.2.0", @"Titanium.UI.KEYBOARD_TYPE_PHONE_PAD");
```
Comment (hansemannn) [inline iphone/Classes/UIModule.m]: It is `UIKeyboardTypePhonePad`, not `UIKeyboardTypeNamePhonePad` for `KEYBOARD_PHONE_PAD`.

**Comment by hansemannn** [comment]
Left comments to be addressed before further review.

**Comment by hieupham007** [comment]
Code reviewed for Android. Looks good

File: apidoc/Titanium/UI/UI.yml [hunk 14]
```diff
@@ -1521,8 +1542,108 @@ properties:
         `KEYBOARD_DEFAULT`.
     type: Number
     permission: read-only
+    deprecated:
+        since: "5.2.0"
+        notes: Use KEYBOARD_TYPE_PHONE_PAD instead.
 
-  - name: KEYBOARD_WEBSEARCH
+  - name: KEYBOARD_URL
```
Comment (hansemannn) [inline apidoc/Titanium/UI/UI.yml]: Why did this property name change?

File: apidoc/Titanium/UI/UI.yml [hunk 15]
```diff
@@ -1521,8 +1542,108 @@ properties:
...
+    permission: read-only
+
+  - name: KEYBOARD_TYPE_EMAIL
+    summary: Use a keyboard suitable for composing email, with the standard keyboard layout.
+    description: |
+        Use with the <Titanium.UI.TextField.keyboardType> and <Titanium.UI.TextArea.keyboardType>
+        properties.
+
+        Note: on Mobile Web, if you enable `passwordType`, the `keyboardType` value will be
+        overridden with `KEYBOARD_TYPE_DEFAULT`.
```
Comment (hansemannn) [inline apidoc/Titanium/UI/UI.yml]: Mobileweb stays `KEYBOARD_TYPE_DEFAULT` for now. @cheekiatng or where should he change those?

**Comment by hansemannn** [comment]
CR + FT passed, PR **approved**!

**Pull request merged**